Riot/Modules/Home/VersionCheck/VersionCheckCoordinator.swift

```diff
@@ -0,0 +1,128 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+private let osVersionToBeDropped = 11
+private let hasOSVersionBeenDropped = false
+
+class VersionCheckCoordinator: VersionCheckBannerViewDelegate, VersionCheckAlertViewControllerDelegate {
+    
+    private let rootViewController: UIViewController
+    private let bannerPresenter: BannerPresentationProtocol
+    private let themeService: ThemeService
+    
+    @UserDefaultsBacked(key: "versionCheckCoordinatorNextDisplayDateTimeInterval")
```

Comment:
It seems that this property should be supported by `RiotSettings` as we need to reset this property after user logout.